python3Packages.setuptools-rust: 0.11.6 -> 0.12.1

danieldk opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Changelog:

https://github.com/PyO3/setuptools-rust/releases/tag/v0.12.1
https://github.com/PyO3/setuptools-rust/releases/tag/v0.12.0

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

danieldk committed:
python3Packages.setuptools-rust: 0.11.6 -> 0.12.1

Changelog:

https://github.com/PyO3/setuptools-rust/releases/tag/v0.12.1
https://github.com/PyO3/setuptools-rust/releases/tag/v0.12.0

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116954` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>home-assistant</li>
</ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/setuptools-rust/default.nix:
```diff
@@ -11,12 +11,12 @@
   pname = "setuptools-rust";
-  version = "0.11.6";
+  version = "0.12.1";
   disabled = isPy27;
```
> Bump minimum Python version to Python 3.6.

Not sure if we should set it to disabled for python < 3.6.

danieldk replied:
Although we don't have Python >3 && < 3.6 anymore, I don't think it hurts. Who knows, maybe someone has an old fork of nixpkgs and is overlaying newer packages ;). I'll update the PR.

danieldk replied:
Fixed. Also fixed the passthru test.

SuperSandro2000 reviewed (approved):
LGTM

I just left some ideas to make improve python rust support.

Would be using this ``https://github.com/PyO3/setuptools-rust/issues/112`` feature make the hook easier?

Maybe cross compilation could benefit from ``https://github.com/PyO3/setuptools-rust/pull/96``?

danieldk commented:
> Would be using this `https://github.com/PyO3/setuptools-rust/issues/112` feature make the hook easier?

Doesn't really affect `setuptools-rust` itself, which doesn't use any hook (outside the normal Python hooks). If projects upload sources to PyPI with vendored crates, then the corresponding derivations should not need `cargoSetupHook`. But that was already true if a project vendored their crates.

> Maybe cross compilation could benefit from `https://github.com/PyO3/setuptools-rust/pull/96`?

Could be. Unfortunately, the last time I tested, cross-compilation failed far before the build would actually a get to a `setuptools-rust` or `maturin` based package. But maybe `master` was just in a bad state at that point.

danieldk committed:
python3Packages.setuptools-rust: fix passthru test

The build wanted to create a Cargo.lock in the top-level source
directory (presumably because it is also a Cargo project), since we
were crating the lock file in sourceRoot (which was set to the example
project). This failed because the sources are read-only. Change the
test to create the lock file in the top-level crate directory, which
is Cargo's default behavior.

danieldk committed:
python3Packages.setuptools-rust: do no mix Python versions in tests

The passthru test used python3Packages, mixing different Python
version when testing on a Python version different that the python3
alias.

SuperSandro2000 commented:
@ofborg test python3Packages.setuptools-rust python38Packages.setuptools-rust python39Packages.setuptools-rust

danieldk commented:
@ofborg build python37Packages.setuptools-rust python38Packages.setuptools-rust python39Packages.setuptools-rust

danieldk commented:
@ofborg build python37Packages.setuptools-rust.passthru.tests python38Packages.setuptools-rust.passthru.tests python39Packages.setuptools-rust.passthru.tests
